### PR TITLE
Account operations

### PR DESCRIPTION
**Pull request by serhatbsts**
the section written to perform withdrawals from the account and deposit money into the account

### CODE REVIEW

File: src/main/java/com/Project/accountManager/controllers/AccountController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @Autowired
+    UserService userService;
+
+    public AccountController(UserService userService, AccountService accountService) {
+        this.userService = userService;
+        this.accountService = accountService;
+    }
+
+    @GetMapping
+    public List<Account> getAllAccount(@RequestParam Optional<Long> userId) {
```
Comment (karaelf33) [inline src/main/java/com/Project/accountManager/controllers/AccountController.java:31]: I see APIs are returning the Account entity.
Let's imagine, the account entity contains sensitive data, such as passwords, confidential investment information, etc...
So is that safe to return all sensitive fields from API?
I highly recommend you research "What is DTO" and use DTO
https://konstantinmb.medium.com/dtos-101-the-what-why-and-how-of-data-transfer-objects-304a03a71a2c

File: src/main/java/com/Project/accountManager/controllers/AccountController.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import com.Project.accountManager.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/account")
+public class AccountController {
```
Comment (karaelf33) [inline src/main/java/com/Project/accountManager/controllers/AccountController.java:18]: God Job 🚀

File: src/main/java/com/Project/accountManager/controllers/AccountController.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Autowired
+    AccountService accountService;
+    @Autowired
```
Comment (karaelf33) [inline src/main/java/com/Project/accountManager/controllers/AccountController.java:22]: you already use Constructure. That's why you don't need to use @Autowried annotation.
Screenshot 2024-05-25 at 08 25 27


so the following will be better -_
  ```
private final AccountService accountService;

    public AccountController(AccountService accountService) {
        this.accountService = accountService;
    }
```
Also, I am suggesting to your research about "Why shouldn't use @Autworied" 
https://www.linkedin.com/pulse/you-should-stop-using-spring-autowired-felix-coutinho

File: src/main/java/com/Project/accountManager/controllers/AccountController.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public Account depositOneAccount(@PathVariable Long accountId, @RequestBody AccountDepositRequest accountDepositRequest) {
+        return accountService.depositOneAccount(accountId, accountDepositRequest);
+    }
+       @PutMapping("/withdrawal/{accountId}")
+    public Account withdrawalAccount(@PathVariable Long accountId, @RequestBody AccountWithdrawalRequest accountWithdrawalRequest) {
+        return accountService.withdrawalAccount(accountId, accountWithdrawalRequest);
+    }
+
+
+
```
Comment (karaelf33) [inline src/main/java/com/Project/accountManager/controllers/AccountController.java:55]: redundant spaces ,
Could you delete it

File: src/main/java/com/Project/accountManager/controllers/UserController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import com.Project.accountManager.entities.User;
+import com.Project.accountManager.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
```
Comment (karaelf33) [inline src/main/java/com/Project/accountManager/controllers/UserController.java:12]: 👏

File: src/main/java/com/Project/accountManager/controllers/AccountController.java (hunk 1 above)
Comment (serhatbsts) [reply src/main/java/com/Project/accountManager/controllers/AccountController.java:31]: OK, necessary arrangements will be made.

File: src/main/java/com/Project/accountManager/controllers/AccountController.java (hunk 3 above)
Comment (serhatbsts) [reply src/main/java/com/Project/accountManager/controllers/AccountController.java:22]: All right. I'll look into it.

File: src/main/java/com/Project/accountManager/controllers/AccountController.java (hunk 4 above)
Comment (serhatbsts) [reply src/main/java/com/Project/accountManager/controllers/AccountController.java:55]: ok

**Pull request merged**